### PR TITLE
feat: implement 0x14 - EQ Opcode 

### PR DESCRIPTION
**Pull request by Alextnetto**
0x14 - EQ opcode
## Pull Request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build-related changes
- [ ] Documentation content changes
- [ ] Other (please describe):

## What is the current behavior?

Resolves: #9 

## What is the new behavior?

- EQ implemented and tested

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: crates/evm/src/instructions/comparison_operations.cairo [hunk 1]
```diff
@@ -40,6 +40,15 @@ impl ComparisonAndBitwiseOperations of ComparisonAndBitwiseOperationsTrait {
...
     fn exec_eq(ref self: ExecutionContext) -> Result<(), EVMError> {
+        let popped = self.stack.pop_n(2)?;
+        let a = *popped[0];
+        let b = *popped[1];
+        let result = if a == b {
+            1
+        } else {
+            0
+        };
+        self.stack.push(result)?;
```
Comment (Eikix) [inline crates/evm/src/instructions/comparison_operations.cairo:51]:
```suggestion
        self.stack.push(result)
```

File: crates/evm/src/instructions/comparison_operations.cairo [hunk 2]
```diff
@@ -40,6 +40,15 @@ impl ComparisonAndBitwiseOperations of ComparisonAndBitwiseOperationsTrait {
...
         Result::Ok(())
```
Comment (Eikix) [inline crates/evm/src/instructions/comparison_operations.cairo:52]: That way you can delete the Result::Ok(())

Comment (Alextnetto) [reply crates/evm/src/instructions/comparison_operations.cairo]: Done, thank @Eikix !

File: crates/evm/src/instructions/comparison_operations.cairo (hunk 1 above)
Comment (Alextnetto) [reply crates/evm/src/instructions/comparison_operations.cairo]: Done!

File: crates/evm/src/instructions/comparison_operations.cairo [hunk 3]
```diff
@@ -40,6 +40,15 @@ impl ComparisonAndBitwiseOperations of ComparisonAndBitwiseOperationsTrait {
     /// 0x14 - EQ
     /// # Specification: https://www.evm.codes/#14?fork=shanghai
     fn exec_eq(ref self: ExecutionContext) -> Result<(), EVMError> {
+        let popped = self.stack.pop_n(2)?;
+        let a = *popped[0];
+        let b = *popped[1];
+        let result = if a == b {
```
[the remark this answers is not in the document]
Comment (Alextnetto) [reply crates/evm/src/instructions/comparison_operations.cairo]: Done! @enitrat. Lmk anything more I can do.

**Review by Eikix — APPROVED** [review]
Nice PR! good job

**Pull request merged**